src/networks/migrations/0008_alter_network_network_type.py

```diff
@@ -0,0 +1,87 @@
+# Generated by Django 5.1.5 on 2025-02-10 14:11
+from django.db import migrations, models
+
+OLD_NETWORK_TYPE = "professional_development_and_skills"
+NEW_NETWORK_TYPE = "professional_networks_and_skills"
+
+
+def update_network_page_and_revision(
+    networks, old_network_type: str, new_network_type: str, Revision, content_type
+):
+    for network_page in networks:
+        if network_page.network_type == old_network_type:
+            network_page.network_type = new_network_type
+            network_page.save(update_fields=["network_type"])
+
+        revisions = Revision.objects.filter(
+            content_type=content_type, object_id=network_page.id
+        )
+        for revision in revisions:
+            if revision.content["network_type"] == old_network_type:
+                revision.content["network_type"] = new_network_type
+                revision.save(update_fields=["content"])
+
+
+def run_update_network_type(apps, old_network_type, new_network_type):
+    Network = apps.get_model("networks", "Network")
+    Revision = apps.get_model("wagtailcore", "Revision")
+
+    ContentType = apps.get_model("contenttypes", "ContentType")
+    content_type = ContentType.objects.get(app_label="networks", model="network")
+
+    update_network_page_and_revision(
+        Network.objects.all(),
+        old_network_type,
+        new_network_type,
+        Revision,
+        content_type,
+    )
```

Comment:
Since the `update_network_page_and_revision` function is only called once, there's no need to separate the logic, we can keep it all in one method.

```suggestion
def run_update_network_type(apps, old_network_type, new_network_type):
    Network = apps.get_model("networks", "Network")
    Revision = apps.get_model("wagtailcore", "Revision")

    ContentType = apps.get_model("contenttypes", "ContentType")
    content_type = ContentType.objects.get(app_label="networks", model="network")

    for network_page in Network.objects.all():
        if network_page.network_type == old_network_type:
            network_page.network_type = new_network_type
            network_page.save(update_fields=["network_type"])

        revisions = Revision.objects.filter(
            content_type=content_type, object_id=network_page.id
        )
        for revision in revisions:
            if revision.content["network_type"] == old_network_type:
                revision.content["network_type"] = new_network_type
                revision.save(update_fields=["content"])
```

This also ends up being less lines of code which is nice 😄

Reply:
Done